python/tests/test_ops.py

```diff
@@ -1665,7 +1665,7 @@ def test_tensordot(self):
 
     def test_inner(self):
         self.assertCmpNumpy([(3,), (3,)], mx.inner, np.inner)
-        self.assertCmpNumpy([(1, 1, 2), (3, 2)], mx.inner, np.inner)
+        self.assertCmpNumpy([(1, 1, 2), (4, 2)], mx.inner, np.inner)
```

Comment:
@jagrit06 something strange here. If I use `3` the test does not work and it gives bogus results. It feels like it's reading from memory that it should not be in the `gemv` specialization. The shapes are `[1, 2]` and `[4, 2].T`.

It's also possible I messed something up, but every other test clears, and benchmarks and examples still work. And this test fails even when I use zero buffers (which don't have any donation). It also passes if I route to our `gemm` instead of `gemv` and it passes if I use MPS `gemm` as well.

Would you mind taking a look at that case?

Reply:
Alright, I'll hop on it

Reply:
Reverted this change since @jagrit06 fixed in #523